org.eclipse.xtext/src/org/eclipse/xtext/build/BuildRequest.java

```diff
@@ -196,4 +200,15 @@ public CancelIndicator getCancelIndicator() {
 	public void setCancelIndicator(CancelIndicator cancelIndicator) {
 		this.cancelIndicator = cancelIndicator;
 	}
+
+	/**
+	 * @since 2.28
+	 */
+	public Set<URI> getSourceLevelUris() {
+		return sourceLevelURIs;		
+	}
+
+	public void setSourceLevelUris(Set<URI> sourceLevelURIs) {
```

Comment:
Please add a comment that this will keep a reference to the given set and subsequent modifications of its content will be observable here and on the call-site.

Reply:
@szarnekow , done